File: public/usage-examples/animations/assign_animation_with_script_named_and_sound-1-example.py
from splashkit import *

anim = create_animation()
script = animation_script_named("WalkingScript")

# Assign by script name + animation name and play sound on first frame if any
assign_animation_with_script_named_and_sound(anim, "WalkingScript", "WalkFront", True)
write_line(f"Assigned WalkFront -> script name: {animation_script_name(script)}")

free_animation(anim)
free_animation_script(script)

File: public/usage-examples/animations/assign_animation_with_script_named_and_sound-1-example.cpp
#include "splashkit.h"

int main()
{
    animation anim = create_animation();
    animation_script script = animation_script_named("WalkingScript");

    // Assign an animation by script name and play sound if requested
    assign_animation(anim, "WalkingScript", "WalkFront", true);
    write_line("Assigned WalkFront -> script name: %s", animation_script_name(script).c_str());

    free_animation(anim);
    free_animation_script(script);
    return 0;
}
